### ISSUE TITLE
Result form and change log: Inconsistencies in trailing zero treatment

### ISSUE THREAD

**Issue by jennyhavah** [mercycorps/TolaActivity#2163]
It's my recollection that we decided on a global rule for trailing zeroes -- to remove them. 

In the case of the result form, I'm noticing trailing zeroes displayed in actual values in two spots:

![image](https://user-images.githubusercontent.com/33670923/73409466-39d49100-42b4-11ea-8d68-011f4d79df51.png)

- [x] 1. Actual value input field: When I add a new result, it automatically rounds if I enter a trailing zero, even before I submit. But when I re-open the result after saving, I see a trailing zero.

- [x] 2. Actual value row in the disaggregation table.

- [x] 3. (Added later) Also noticed a trailing zero issue in the change log. I can't tell if that's connected to this issue, but seemed like as good a place as any to put it:

![image](https://user-images.githubusercontent.com/33670923/74198353-febc5100-4c15-11ea-8449-d563a815de2a.png)

**Comment by camgnostic** [comment]
On hold pending disposition of #2125 to avoid reduplication of work

**Comment by camgnostic** [comment]
Result form section complete, assigning to sanjuro and moving back to ready to handle change log element

**Comment by sanjuroj** [comment]
@jennyhavah One testing note is that older data will still be inconsistent (the extra zeros are saved to the database). But any new changelog data created should have extra zeros shaved off the end.

**Comment by jennyhavah** [comment]
For new results, no longer seeing trailing zeroes when I reopen a saved result, or in the change log.

**Comment by jennyhavah** [comment]
Sanjuro noticed that this issue seems to be happening again -- possibly a regression?

**Comment by sanjuroj** [comment]
The problem was that hitting the reset button was turning e.g. 6 into 6.00 in both the actual value and disagg fields.  This behavior should be corrected now.

**Comment by jennyhavah** [comment]
Found a couple of issues.

1) I'm still seeing trailing zeroes in the change log when the value is zero. (No trailing zeroes when the value is greater than zero.)

![image](https://user-images.githubusercontent.com/33670923/77716280-0011c500-6f9b-11ea-8ac8-faa52828753f.png)

2) While reproducing the above issue, I discovered that I can no longer submit a result value of zero. Disaggregated values can be zero, but not the total. A result of zero should be allowable. (It is allowed as expected on Prod.)

![image](https://user-images.githubusercontent.com/33670923/77716340-2d5e7300-6f9b-11ea-9d0e-b4fa0d3d6ed8.png)
